src/components/slidePanel/createSlidePanel.ts

```diff
@@ -0,0 +1,205 @@
+		},
+
+		onSwipeMove(this: SlidePanel, event: MouseEvent & TouchEvent) {
+			const state = internalStateMap.get(this);
+
+			// Ignore mouse movement when not clicking
+			if (!state.swiping) {
+				return;
+			}
+
+			const {
+				width = 256,
+				align = 'left'
+			} = this.properties;
+
+			// Current pointer position
+			let currentX = event.type === 'touchmove' ? event.changedTouches[0].screenX : event.pageX;
+			// Difference between current and initial pointer position
+			let delta = align === 'right' ? currentX - state.initialX : state.initialX - currentX;
+			// Transform to apply
+			state.transform = 100 * delta / width;
+
+			internalStateMap.set(this, state);
+
+			// Prevent panel from sliding past screen edge
+			if (delta <= 0) {
+				return;
+			}
+
+			// Move the panel
+			if (state.content) {
+				state.content.style.transform =  `translateX(${ align === 'left' ? '-' : '' }${ state.transform }%)`;
+			}
+		},
+
+		onSwipeEnd(this: SlidePanel, event: MouseEvent & TouchEvent) {
+			const state = internalStateMap.get(this);
+			state.swiping = false;
+
+			const {
+				width = 256,
+				align = 'left',
+				onRequestClose
+			} = this.properties;
+
+			// Current pointer position
+			let currentX = event.type === 'touchend' ? event.changedTouches[0].screenX : event.pageX;
+			// Difference between current and initial pointer position
+			let delta = align === 'right' ? currentX - state.initialX : state.initialX - currentX;
+
+			// If the panel was swiped far enough to close
+			if (delta > width / 2) {
+				// Cache the transform to apply on next render
+				state.transform = 100 * delta / width;
+				internalStateMap.set(this, state);
+				onRequestClose && onRequestClose();
+			}
+			// If the underlay was clicked
+			else if (delta > -5 && delta < 5 && (<HTMLElement> event.target).classList.contains(css.underlay)) {
+				internalStateMap.set(this, state);
+				onRequestClose && onRequestClose();
+			}
+			// If panel was not swiped far enough to close
+			else if (delta > 0) {
+				// Animate the panel back open
+				state.content && state.content.classList.add(css.slideIn);
+				internalStateMap.set(this, state);
+			}
+		},
+
+		afterCreate(this: SlidePanel, element: HTMLElement) {
+			const state = internalStateMap.get(this);
+			element.addEventListener('transitionend', this.onTransitionEnd!);
+			state.content = element;
+			internalStateMap.set(this, state);
+		},
+
+		onTransitionEnd(this: SlidePanel, event: TransitionEvent) {
+			const content = (<HTMLElement> event.target);
+			content.classList.remove(css.slideIn, css.slideOut);
+			content.style.transform = '';
+		},
+
+		render(this: SlidePanel): DNode {
+			const state = internalStateMap.get(this);
+			const {
+				open = false,
+				align = 'left',
+				underlay = false,
+				onOpen
+			} = this.properties;
+
+			const classes: {[key: string]: any} = {};
+			const styles: {[key: string]: any} = {};
+
+			classes[css.content] = true;
```

Comment:
I think that we should be using the `this.class(` function as it effectively manages the creation of any required `false` classes, so for this example you could just have an array of classNames that you want to be set and then spread them `this.classes(...classesToSet).get()`?

```ts
const classes = [css.content];

if (open && !state.wasOpen) {
    classes.push(css.slideIn);
}
else if (!open && state.wasOpen) {
    classes.push(css.slideOut);
    if (state.transform !== 0) {
        styles['transform'] = `translateX(${ align === 'left' ? '-' : '' }${ state.transform }%)`;
    }
}

const content = v('div', { 
    key: 'content', 
    classes:  this.classes(...classes).get(), 
    styles, 
    afterCreate: this.afterCreate 
}, this.children);
```